index.js

```diff
@@ -29,7 +29,7 @@ module.exports = function kindOf(val) {
 
   // functions
   if (type === 'function' || val instanceof Function) {
-    if (val.constructor.name.slice(0, 9) === 'Generator') {
+    if (typeof val.constructor.name !== 'undefined' && val.constructor.name.slice(0, 9) === 'Generator') {
```

Comment:
when so much bytes is used, it's probably better to directly be checking of `string`.

So:

```js
var name = (val.constructor && val.constructor.name) || ''
if (name && name.length && name.slice(0, 9) === 'Generator') {
```